pkg/probe/http/request.go

```diff
@@ -113,7 +113,7 @@ func formatURL(scheme string, host string, port int, path string) *url.URL {
 func v1HeaderToHTTPHeader(headerList []v1.HTTPHeader) http.Header {
 	headers := make(http.Header)
 	for _, header := range headerList {
-		headers[header.Name] = append(headers[header.Name], header.Value)
+		headers.Add(header.Name, header.Value)
```

Comment:
```suggestion
		headers.Add(http.CanonicalHeaderKey(header.Name), header.Value)
```
I think this should be keeping using CanonicalHeaderKey here.

Reply:
It's handled inside the `Add` method. The test is passed

Reply:
```
func (h MIMEHeader) Add(key, value string) {
	key = CanonicalMIMEHeaderKey(key)
	h[key] = append(h[key], value)
}
```
Got it. Sorry for the noise.
/lgtm